packages/polymorphic/src/Polymorph.tsx

```diff
@@ -0,0 +1,34 @@
+import React from 'react';
+import { usePolymorphicComponent } from './Polymorphic.hooks';
+import {
+  PolymorphicAs,
+  PolymorphicComponentType,
+  PolymorphicPropsWithRef,
+  PolymorphicRef,
+} from './Polymorphic.types';
+
+/**
+ * Base polymorphic component
+ */
+export const BasePolymorph = <T extends PolymorphicAs = 'div'>(
+  { as, children, ...rest }: PolymorphicPropsWithRef<T>,
+  ref: PolymorphicRef<T>,
+) => {
+  const Component = usePolymorphicComponent(as);
+
+  return (
+    <Component {...rest} ref={ref}>
+      {children}
+    </Component>
+  );
+};
+
+/**
+ * Polymorphic component that supports forwarded refs
+ *
+ * With the `as` prop, Polymorphic can dynamically render
+ * as any HTML element or React Component.
+ */
+export const Polymorph: PolymorphicComponentType =
+  React.forwardRef(BasePolymorph);
+Polymorph.displayName = 'Polymorph';
```

Comment:
are we still supporting propTypes?

Reply:
We've gone back & forth on that. We should bring this up at the next engineers jam

Reply:
Yeah I think we'd need to support it for devs that don't use typescript

Reply:
👍 Updated types & Added tests to ensure propTypes support